### PR TITLE
add nbt path.

### PR DESCRIPTION
**Pull request by ch-yx**
add nbt path.
need optimizing.

### CODE REVIEW

File: nbtlib/path.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            return type(self)(self.parts+(key,))
+        if isinstance(key,str):
+            return type(self)(self.parts+((key,None),))
+        if isinstance(key,Compound):
+            *p,q=self.parts
+            q,r=q
+            r=r or Compound()
+            r=Compound(r.copy())
+            r.update(key)
+            return type(self)((*p,(q,r)))
```
Comment (ch-yx) [inline nbtlib/path.py]: *p,(q,r)=self.parts
r=r or {}
r=Com...

File: nbtlib/path.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    tag = parser.parse()
+
+    cursor = parser.token_span[1]
+    leftover = literal[cursor:]
+
+    return tag,leftover
+
+
+def match(obj,pattern):
+	for i in Int.all_tags.values():
```
Comment (SPGoding) [inline nbtlib/path.py]: Style issue: I don't recommend to use both tabs and spaces for indent in one file. It seems that the owner prefers 4 spaces.

**Review by SPGoding — CHANGES_REQUESTED** [review]

File: nbtlib/path.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+            if isinstance(item,int):
+                out_=[]
+                for i in (i for i in out if isinstance(i,List)):
+                    try:
+                        out_.append(i[item])
+                    except:
+                        pass
+                out=out_
+            elif item is None:
+                out=sum((i for i in out if isinstance(i,List)),[])
```
Comment (ch-yx) [inline nbtlib/path.py]: I d better using filter.

Comment (ch-yx) [reply nbtlib/path.py]: not compatible with (byte/int/long)arrays!
I have to fix it!

File: nbtlib/path.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        if isinstance(that,str):
+            that=type(this)(that)
+        if that.parts==((None,None),):
+            return this
+        if isinstance(that.parts[0],(int,type(None))) or that.parts[0][0] is not None:
+            return type(this)(this.parts+that.parts)
+        *a,(b,c)=this
+        (d,e),*f=that
+        c=c or {}
+        c=Compound(c.copy())
```
Comment (ch-yx) [inline nbtlib/path.py]: should change to copy.deepcopy

Comment (ch-yx) [reply nbtlib/path.py]: done

File: nbtlib/path.py (hunk 3 above)
Comment (ch-yx) [reply nbtlib/path.py]: done.

File: nbtlib/path.py (hunk 2 above)
Comment (ch-yx) [reply nbtlib/path.py]: done.

**Comment by vberlier** [comment]
Thanks for the pull request, I finally managed to review it! It would need a bit more work but I'm not convinced my initial idea is that good anymore. I'm going to explain a bit more in the issue #34 but I took some time to think about it and I won't be merging this because I came up with an alternative implementation.

**Pull request closed without merging**